### PR TITLE
Equipment generator bow quiver

### LINKED ISSUE #495: Equipment Generator Bow with Quiver
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Currently a quiver is not guaranteed when you spawn with a bow thus having no arrows.

## Intended Design

Always provide a quiver to the player in the case that they spawn with a bow

## Location

\Missions\Services\Arena\RandomEquipmentGenerator.cs

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by Magenstor**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] All new classes have class-level documentation comments, if there are any at all
- [X] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

Sometimes arrows ammo and throwing weapons dont spawn because there is some items in the armor/weapons dictionary that cannot spawn in players, i just remove the items from the dictionary.

## What is the new behaviour?

Resolves #

## Other information

### CODE REVIEW

**Review by EgardA — DISMISSED** [review]
I'm fine with the hardcoding as this will only be used for testing, anyone disagree?

File: source/Missions/Services/Arena/RandomEquipmentGenerator.cs [hunk 1]
```diff
@@ -46,17 +47,30 @@ internal class RandomEquipmentGenerator : IRandomEquipmentGenerator
...
                 {
                     result.Add(item.ItemType, new List<ItemObject>());
                 }
+
                 result[item.ItemType].Add(item);
-            }
 
+                //Here im harcoding the elements that are causing the arrows not to spawn and the same for the throwing weapons
+                if (
+                item.StringId == "ballista_projectile_burning" || item.StringId == "ballista_projectile" || item.StringId == "throwing_stone" || item.StringId == "boulder" ||
```
Comment (garrettluskey) [inline source/Missions/Services/Arena/RandomEquipmentGenerator.cs]: Convert this to a static HashSet and add this as a check to line 52

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Missions/Services/Arena/RandomEquipmentGenerator.cs [hunk 2]
```diff
@@ -46,17 +47,30 @@ internal class RandomEquipmentGenerator : IRandomEquipmentGenerator
 
             foreach (var item in allItems)
             {
+
                 bool keyExists = result.ContainsKey(item.ItemType);
                 if (!keyExists)
```
Comment (garrettluskey) [inline source/Missions/Services/Arena/RandomEquipmentGenerator.cs:58]:
```suggestion
                if (<HashSet>.Contains(item.StringId) continue;
                if (!keyExists)
```

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Missions/Services/Arena/RandomEquipmentGenerator.cs [hunk 3]
```diff
@@ -43,20 +45,28 @@ internal class RandomEquipmentGenerator : IRandomEquipmentGenerator
         {
             IEnumerable<ItemObject> allItems = Game.Current.ObjectManager.GetObjectTypeList<ItemObject>();
             IDictionary<ItemTypeEnum, List<ItemObject>> result = new Dictionary<ItemTypeEnum, List<ItemObject>>();
+            //Here im harcoding the elements that are causing the arrows not to spawn and the same for the throwing weapons
+            HashSet<string> deleteItems = new HashSet<string>{"ballista_projectile_burning", "ballista_projectile", "throwing_stone", "boulder",
```
Comment (garrettluskey) [inline source/Missions/Services/Arena/RandomEquipmentGenerator.cs]: move this to the class scope `private static readonly...`
Rename this to `ExcludedItems`

Should be perfect after that

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Missions/Services/Arena/RandomEquipmentGenerator.cs (hunk 3 above)
Comment (Magenstor) [reply source/Missions/Services/Arena/RandomEquipmentGenerator.cs]: ye my bad, i forgot

**Review by garrettluskey — APPROVED** [review]

**Pull request merged**